index.d.ts

```diff
@@ -0,0 +1,18 @@
+import {Middleware, Dispatch} from "redux";
+
+
+export type ThunkAction<R, S, E> = (dispatch: Dispatch<S>, getState: () => S,
+                                    extraArgument: E) => R;
+
+declare module "redux" {
+  export interface Dispatch<S> {
+    <R, E>(asyncAction: ThunkAction<R, S, E>): R;
```

Comment:
Of course, it does. Your `Dispatch` is tied with your store which is typed by state type. With your signature one could redefine the state type, which would not be allowed in reality. I have the same concern with `E` extra argument type, but cannot be well defined with current design. I've been experimenting with some new approach that could address this issue as well.